### PR TITLE
Added/Configured a code analyzer

### LINKED ISSUE #280: [Enhancement] Add analyzers
Add analyzers into the project.

### PR DESCRIPTION
**Pull request by pictos**
### Description of Change ###

Describe your changes here. 

### Bugs Fixed ###

- Fixes #280

Provide links to issues here. Ensure that a GitHub issue was created for your feature or bug fix before sending PR.

### Behavioral Changes ###

Now we have warnings if the code doesn't follow our code style. But it needs to be configured.
image

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [ ] Has samples (if omitted, state reason in description)
- [x] Rebased on top of main at time of PR
- [ ] Changes adhere to coding standard
- [ ] Updated documentation

### CODE REVIEW

**Comment by pictos** [comment]
The build failed because a unit test you can see the [log here](https://dev.azure.com/xamarin/public/_build/results?buildId=26302&view=logs&j=7868e7c0-c859-51b7-c3be-08c48f6833d0&t=fb76b7ed-e4f5-522c-a151-eb3e57e0f1a4&l=261) I don't know if it's related to this PR

File: XamarinCommunityToolkit/Xamarin.CommunityToolkit.csproj [hunk 1]
```diff
@@ -31,7 +31,6 @@
     <PackageProjectUrl>https://github.com/xamarin/XamarinCommunityToolkit</PackageProjectUrl>
     <DebugType>portable</DebugType>
     <Configurations>Debug;Release</Configurations>
-    <LangVersion>8.0</LangVersion>
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkit/Xamarin.CommunityToolkit.csproj:34]: Why do we delete this line?)

File: XamarinCommunityToolkitSample/Xamarin.CommunityToolkit.Sample.csproj [hunk 2]
```diff
@@ -3,7 +3,6 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <ProduceReferenceAssembly>true</ProduceReferenceAssembly>
-    <LangVersion>8.0</LangVersion>
```
Comment (AndreiMisiukevich) [inline XamarinCommunityToolkitSample/Xamarin.CommunityToolkit.Sample.csproj:6]: And this

**Comment by sthewissen** [comment]
> The build failed because a unit test you can see the [log here](https://dev.azure.com/xamarin/public/_build/results?buildId=26302&view=logs&j=7868e7c0-c859-51b7-c3be-08c48f6833d0&t=fb76b7ed-e4f5-522c-a151-eb3e57e0f1a4&l=261) I don't know if it's related to this PR. It passes locally.

That specific test does indeed not have anything to do with this. I believe that that specific test is time-based, so perhaps something being a bit slower than usual is causing it to fail.

File: XamarinCommunityToolkit/Xamarin.CommunityToolkit.csproj (hunk 1 above)
Comment (sthewissen) [reply XamarinCommunityToolkit/Xamarin.CommunityToolkit.csproj:34]: It's moved into `Directory.Build.props` it seems?

**Review by sthewissen — DISMISSED** [review]

File: Directory.Build.props [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+ <PropertyGroup>
```
Comment (AndreiMisiukevich) [inline Directory.Build.props:4]:
```suggestion
  <PropertyGroup>
```

**Review by AndreiMisiukevich — APPROVED** [review]

**Review by sthewissen — APPROVED** [review]

**Pull request merged**